src/tools/allocator-recipe-resolver.ts

```diff
@@ -216,6 +216,11 @@ export class AllocatorRecipeResolver {
     }
     return this.createHandleRegistry.get(handle);
   }
+
+  /** Returns set of recipes with unique names. */
+  private uniqueRecipes(recipes: Recipe[]): Recipe[] {
+    return recipes.filter((r, idx, self) => idx === self.findIndex((x) => x.name === r.name));
```

Comment:
instead of names, can we use something that's less prone to errors, e.g. recipe.toString() representation?